engine/crates/engine-v2/src/execution/planner/mod.rs

```diff
@@ -1,131 +1,217 @@
 mod builder;
 mod query_modifier;
 
-use std::sync::Arc;
+use std::{mem::take, sync::Arc};
 
+use builder::ExecutionBuilder;
+use id_newtypes::IdRange;
 use itertools::Itertools;
 
-use builder::RequirementsBuildContext;
-
 use crate::{
-    execution::{ExecutionPlan, ExecutionPlanId, PlanningResult, PreExecutionContext},
+    execution::{ExecutionPlan, ExecutionPlanId, PlanningResult, PreExecutionContext, ResponseModifierExecutorId},
     operation::{FieldId, LogicalPlanId, PreparedOperation, Variables},
     response::{ResponseViewSelection, ResponseViews},
     sources::PreparedExecutor,
     utils::BufferPool,
     Runtime,
 };
 
-use super::{header_rule::create_subgraph_headers_with_rules, ExecutableOperation};
+use super::{header_rule::create_subgraph_headers_with_rules, ExecutableOperation, ResponseModifierExecutor};
 
 pub(super) struct ExecutionPlanner<'ctx, 'op, R: Runtime> {
     ctx: &'op PreExecutionContext<'ctx, R>,
     operation: ExecutableOperation,
+    build_context: BuildContext,
+}
+
```

Comment:
What's the point/difference between this and the ExecutionBuilder I just finished reading?  I can only see 3 functions on this thing - could they just be free standing functions that take 1-3 parameters?  There's a ton of ambiguous structs in this code, it'd be nice if we could ditch some of them...

Reply:
Well yeah, my problem is there I currently need to modify the `ExecutableOperation` at some point, and then I'm relying on it being immutable and I'm alternating those phases... So that's something I want to fix, and seems like I fixed the core issue that I had at some point. Those could be free-standing functions. I will take a stab on this next week